### PR TITLE
QSP-1: Fixed protocol fees withdrawal

### PR DESCRIPTION
**Pull request by AnonFinDev**

### CODE REVIEW

File: contracts/protocol/clearinghouse/ClearingHouse.sol [hunk 1]
```diff
@@ -147,11 +148,15 @@ contract ClearingHouse is
...
-        for (uint256 i = 0; i < wrapperAddresses.length; i++) {
-            uint256 wrapperFee = IVPoolWrapper(wrapperAddresses[i]).collectAccruedProtocolFee();
-            emit Account.ProtocolFeesWithdrawn(wrapperAddresses[i], wrapperFee);
+        for (uint256 i = 0; i < poolIds.length; i++) {
+            //check if the supplied address is a deployed vpoolwrapper
+            uint32 poolId = poolIds[i];
+            IVPoolWrapper poolWrapper = protocol.pools[poolId].vPoolWrapper;
+            if (address(poolWrapper).isZero()) revert InvalidPoolId(poolId);
+            uint256 wrapperFee = poolWrapper.collectAccruedProtocolFee();
+            emit Account.ProtocolFeesWithdrawn(poolId, wrapperFee);
```
Comment (134dd3v) [inline contracts/protocol/clearinghouse/ClearingHouse.sol:159]: We can have this event in ClearingHouse, no?

Comment (AnonFinDev) [reply contracts/protocol/clearinghouse/ClearingHouse.sol:159]: eeb8076a9519f7a349af376bd61694c8f4d95aec

File: contracts/protocol/clearinghouse/ClearingHouse.sol [hunk 2]
```diff
@@ -50,6 +50,7 @@ contract ClearingHouse is
 
     error NotRageTradeFactory();
     error ZeroAmount();
+    error InvalidPoolId(uint32 poolId);
```
Comment (134dd3v) [inline contracts/protocol/clearinghouse/ClearingHouse.sol:53]: We need to move this to IClearingHouseErrors in some PR.

Comment (AnonFinDev) [reply contracts/protocol/clearinghouse/ClearingHouse.sol]: c8f6bb9257fa37858f8c629e2ff7baad8253e88e

**Pull request closed without merging**